packages/atomic/src/components/result-template-components/atomic-result-value/atomic-result-value.tsx

```diff
@@ -12,17 +12,38 @@ import {
 export class AtomicResultValue {
   @Element() host!: HTMLDivElement;
   @Prop() value = '';
+  @Prop() shouldHighlightWith = '';
 
   @ResultContext() private result!: Result;
 
+  /**
+   * Returns the value of 'shouldHighlightWith' prop from the Result
+   */
+  private getHighlightsFromResult() {
+    const highlightsKeywords = Object.entries(this.result).find(
+      (keyValue) => keyValue[0] === this.shouldHighlightWith
+    );
+    return highlightsKeywords![1] as string[];
+  }
+
   @ResultContextRenderer
-  public render() {
-    const resultValue = ResultTemplatesHelpers.getResultProperty(
+  render() {
+    let resultValue = ResultTemplatesHelpers.getResultProperty(
       this.result,
       this.value
     );
+
     if (resultValue !== null) {
-      return resultValue;
+      if (this.shouldHighlightWith && this.shouldHighlightWith in this.result) {
+        resultValue = highlightString({
+          content: resultValue as string,
```

Comment:
Why is this cast needed? Same for line 44.

Reply:
`resultValue` can be any property of `Result`. If we check that it is a string, we would need additional code to render when we get a property that is a `number`, `boolean`, etc. But, I'm not sure if there is a use case when we would render something from `Result` that is not a `string`. In that case, checking it is not `null` is fine. Otherwise, checking it is a `string` can be used.